jcl/src/jdk.management/share/classes/com/openj9/lang/management/OpenJ9DiagnosticsMXBean.java

```diff
@@ -0,0 +1,148 @@
+	 * handled.
+	 *
+	 * @throws DumpConfigurationUnavailableException if the configuration cannot be changed because a dump is already in progress.
+         * @throws SecurityException if there is a security manager and it doesn't allow the checks required to change the dump settings
+     	 */
+	public void resetDumpOptions() throws DumpConfigurationUnavailableException;
+
+	/**
+	 * This function sets options for the dump subsystem.
+	 * The dump option is passed in as a String. Use the same syntax as the -Xdump command-line option, with the 
+	 * initial -Xdump: omitted. See Using the -Xdump option as described in the section on dump agents in the 
+	 * documentation for the OpenJ9 JVM. This method may throw a DumpConfigurationUnavailableException if the dump
+	 * configuration cannot be altered.
+	 *
+	 * @param dumpOptions the options string to be set
+	 * @throws InvalidDumpOptionException if the specified option cannot be set or is incorrect.
+	 * @throws DumpConfigurationUnavailableException if the configuration cannot be changed because a dump is already in progress.
+	 * @throws SecurityException if there is a security manager and it doesn't allow the checks required to change the dump settings
+	 * @throws NullPointerException if options is null
+	 */
+	public void setDumpOptions(String dumpOptions) throws InvalidDumpOptionException, DumpConfigurationUnavailableException;
+
+	/**
+	 * This function triggers the specified dump agent. Dump Agents supported - java, snap, system, heap
+	 * A java dump is in a human-readable format, and summarizes the state of the JVM
+         * The default heap dump format (a phd file) is not human-readable.
+         * A system dump is a platform-specific file that contains information about the active processes, threads, and
+	 * system memory. System dumps are usually large. 
+	 * The snap dump format is not human-readable and must be processed using the trace formatting tool supplied  with the OpenJ9 JVM.
+	 *
+	 * @param dumpAgent the dump agent to be triggered
+	 * @throws IllegalArgumentException if the specified dump agent is invalid or unsupported by this method.
+	 * @throws RuntimeException if the vm does not contain RAS dump support
+	 * @throws SecurityException if there is a security manager and it doesn't allow the checks required to trigger this dump
+	 */
+	public void triggerDump(String dumpAgent) throws IllegalArgumentException;
+
+	/**
+	 * This function triggers the specified dump agent. Dump Agents supported - java, snap, system, heap.
+	 * The JVM will attempt to write the file to the specified file name. This may
+	 * include replacement tokens as documented in the section on dump agents
+	 * in the documentation for the OpenJ9 JVM.
+	 * 
+	 * A string containing the actual file name written to is returned. This may not
+	 * be the same as the requested filename for several reasons:
+	 * <ul>
+	 * <li>null or the empty string were specified, this will cause the JVM to write the
```

Comment:
This should use 'was', not 'were': there *was* only one past as this case applies.